### ISSUE TITLE
Ruby-Lsp Server no longer starts - Version error

### ISSUE THREAD

**Issue by dmrwebdev** [Shopify/ruby-lsp#2891]
### Description

A couple hours ago the ruby-lsp extension suddenly stopped working for me with the following warnings:

`Ruby LSP client: couldn't create connection to server.`
`Pending response rejected since connection got disposed`
`Launching the Ruby LSP failed. This typically happens due to an error with version manager integration or Bundler issues.`

No changes to configuration were made and extension was working fine all up day up until this point. Timing wise it lines up with when the Github action on main bumped the ruby-lsp version to 0.22.1, but I am not sure if this is a cause or not.

#### VS Code Version

1.95.3

#### Ruby LSP Extension Version

0.8.13

#### Ruby LSP Server Version

Unknown

#### Ruby Version

3.3.5

#### Ruby Version Manager

rvm

#### Installed Extensions

<details>
<summary>Click to expand</summary>

- ruby-lsp (0.8.13)
</details>

#### Ruby LSP Settings

<details>
<summary>Click to expand</summary>

##### Workspace

```json
{}
```

##### User

```json
{
  "enableExperimentalFeatures": false,
  "enabledFeatures": {
    "codeActions": true,
    "diagnostics": true,
    "documentHighlights": true,
    "documentLink": true,
    "documentSymbols": true,
    "foldingRanges": true,
    "formatting": true,
    "hover": true,
    "inlayHint": true,
    "onTypeFormatting": true,
    "selectionRanges": true,
    "semanticHighlighting": true,
    "completion": true,
    "codeLens": true,
    "definition": true,
    "workspaceSymbol": true,
    "signatureHelp": true,
    "typeHierarchy": true,
    "references": true
  },
  "featuresConfiguration": {},
  "addonSettings": {},
  "rubyVersionManager": {
    "identifier": "auto"
  },
  "customRubyCommand": "",
  "formatter": "auto",
  "linters": null,
  "bundleGemfile": "",
  "testTimeout": 30,
  "branch": "main",
  "pullDiagnosticsOn": "both",
  "useBundlerCompose": false,
  "bypassTypechecker": false,
  "rubyExecutablePath": "",
  "indexing": {},
  "erbSupport": true,
  "useLauncher": false,
  "featureFlags": {}
}
```

</details>

### Reproduction steps

Simply reload window.

**Code snippet or error message**
```
2024-11-20 15:33:50.172 [info] (trigo-app) Ruby LSP> Running bundle install for the composed bundle. This may take a while...
Ruby LSP> Command: (bundle _2.5.9_ check || bundle _2.5.9_ install) 1>&2

2024-11-20 15:33:50.288 [info] (trigo-app) The git source https://github.com/Shopify/ruby-lsp.git is not yet checked out.
Please run `bundle install` before trying to start your application

2024-11-20 15:33:50.427 [info] (trigo-app) Fetching https://github.com/Shopify/ruby-lsp.git

2024-11-20 15:33:51.447 [info] (trigo-app) Fetching gem metadata from https://rubygems.org/
2024-11-20 15:33:51.447 [info] (trigo-app) .
2024-11-20 15:33:51.554 [info] (trigo-app) .
2024-11-20 15:33:51.591 [info] (trigo-app) .
2024-11-20 15:33:51.632 [info] (trigo-app) .
2024-11-20 15:33:51.648 [info] (trigo-app) .
2024-11-20 15:33:51.655 [info] (trigo-app) .
2024-11-20 15:33:51.658 [info] (trigo-app) .
2024-11-20 15:33:51.660 [info] (trigo-app) .
2024-11-20 15:33:51.660 [info] (trigo-app) .
2024-11-20 15:33:52.142 [info] (trigo-app) 

2024-11-20 15:33:52.184 [info] (trigo-app) Resolving dependencies...

2024-11-20 15:33:53.098 [info] (trigo-app) Could not find compatible versions

Because ruby-lsp-rails < 0.2.0 depends on ruby-lsp ~> 0.4.5
  and ruby-lsp-rails >= 0.2.0, < 0.2.3 depends on ruby-lsp >= 0.6.2, < 0.8.0,
  ruby-lsp-rails < 0.2.3 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.8.0.
And because ruby-lsp-rails >= 0.2.3, < 0.2.4 depends on ruby-lsp >= 0.8.0, <
0.9.0,
  ruby-lsp-rails < 0.2.4 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0.
And because ruby-lsp-rails >= 0.2.4, < 0.2.5 depends on ruby-lsp >= 0.9.1, <
0.10.0
  and ruby-lsp-rails >= 0.2.5, < 0.2.6 depends on ruby-lsp >= 0.10.0, < 0.11.0,
ruby-lsp-rails < 0.2.6 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.11.0.
And because ruby-lsp-rails >= 0.2.6, < 0.2.7 depends on ruby-lsp >= 0.11.0, <
0.12.0
  and ruby-lsp-rails >= 0.2.7, < 0.2.8 depends on ruby-lsp >= 0.12.0, < 0.13.0,
ruby-lsp-rails < 0.2.8 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.13.0.
And because ruby-lsp-rails >= 0.2.8, < 0.2.10 depends on ruby-lsp >= 0.13.0, <
0.14.0
  and ruby-lsp-rails >= 0.2.10, < 0.3.1 depends on ruby-lsp >= 0.14.0, < 0.15.0,
ruby-lsp-rails < 0.3.1 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.15.0.
And because ruby-lsp-rails >= 0.3.1, < 0.3.4 depends on ruby-lsp >= 0.14.2, <
0.15.0
  and ruby-lsp-rails >= 0.3.4, < 0.3.6 depends on ruby-lsp >= 0.16.0, < 0.17.0,
ruby-lsp-rails < 0.3.6 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.15.0 OR >= 0.16.0, < 0.17.0.
And because ruby-lsp-rails >= 0.3.6, < 0.3.7 depends on ruby-lsp >= 0.16.5, <
0.17.0
  and ruby-lsp-rails >= 0.3.7, < 0.3.8 depends on ruby-lsp >= 0.17.0, < 0.18.0,
ruby-lsp-rails < 0.3.8 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.15.0 OR >= 0.16.0, < 0.18.0.
And because ruby-lsp-rails >= 0.3.8, < 0.3.12 depends on ruby-lsp >= 0.17.2, <
0.18.0
and ruby-lsp-rails >= 0.3.12, < 0.3.16 depends on ruby-lsp >= 0.17.12, <
0.18.0,
ruby-lsp-rails < 0.3.16 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.15.0 OR >= 0.16.0, < 0.18.0.
And because ruby-lsp-rails >= 0.3.16, < 0.3.17 depends on ruby-lsp >= 0.18.0, <
0.19.0
and ruby-lsp-rails >= 0.3.17, < 0.3.19 depends on ruby-lsp >= 0.19.0, <
0.20.0,
ruby-lsp-rails < 0.3.19 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.15.0 OR >= 0.16.0, < 0.20.0.
And because ruby-lsp-rails >= 0.3.19, < 0.3.22 depends on ruby-lsp >= 0.20.0, <
0.21.0
and ruby-lsp >= 0.4.5, < 0.5.A OR >= 0.6.2, < 0.9.0 OR >= 0.9.1, < 0.15.0 OR
>= 0.16.0, < 0.21.0 could not be found in
https://github.com/Shopify/ruby-lsp.git (at main@fc2d91c),
  ruby-lsp-rails < 0.3.22 cannot be used.
So, because ruby-lsp-rails >= 0.3.22 depends on ruby-lsp >= 0.21.2, < 0.22.0
and ruby-lsp >= 0.21.2, < 0.22.0 could not be found in
https://github.com/Shopify/ruby-lsp.git (at main@fc2d91c),
  version solving has failed.

2024-11-20 15:33:53.115 [info] (trigo-app) Ruby LSP> Running bundle install failed. Trying to re-generate the composed bundle from scratch

2024-11-20 15:33:53.116 [info] (trigo-app) Ruby LSP> Running bundle install for the composed bundle. This may take a while...
Ruby LSP> Command: (bundle _2.5.9_ check || bundle _2.5.9_ install) 1>&2

2024-11-20 15:33:53.231 [info] (trigo-app) The git source https://github.com/Shopify/ruby-lsp.git is not yet checked out.
Please run `bundle install` before trying to start your application

2024-11-20 15:33:53.373 [info] (trigo-app) Fetching https://github.com/Shopify/ruby-lsp.git

2024-11-20 15:33:54.444 [info] (trigo-app) Fetching gem metadata from https://rubygems.org/
2024-11-20 15:33:54.444 [info] (trigo-app) .
2024-11-20 15:33:54.553 [info] (trigo-app) .
2024-11-20 15:33:54.590 [info] (trigo-app) .
2024-11-20 15:33:54.631 [info] (trigo-app) .
2024-11-20 15:33:54.647 [info] (trigo-app) .
2024-11-20 15:33:54.654 [info] (trigo-app) .
2024-11-20 15:33:54.658 [info] (trigo-app) .
2024-11-20 15:33:54.659 [info] (trigo-app) .
2024-11-20 15:33:54.660 [info] (trigo-app) .
2024-11-20 15:33:55.143 [info] (trigo-app) 

2024-11-20 15:33:55.184 [info] (trigo-app) Resolving dependencies...

2024-11-20 15:33:56.082 [info] (trigo-app) Could not find compatible versions

Because ruby-lsp-rails < 0.2.0 depends on ruby-lsp ~> 0.4.5
  and ruby-lsp-rails >= 0.2.0, < 0.2.3 depends on ruby-lsp >= 0.6.2, < 0.8.0,
  ruby-lsp-rails < 0.2.3 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.8.0.
And because ruby-lsp-rails >= 0.2.3, < 0.2.4 depends on ruby-lsp >= 0.8.0, <
0.9.0,
  ruby-lsp-rails < 0.2.4 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0.
And because ruby-lsp-rails >= 0.2.4, < 0.2.5 depends on ruby-lsp >= 0.9.1, <
0.10.0
  and ruby-lsp-rails >= 0.2.5, < 0.2.6 depends on ruby-lsp >= 0.10.0, < 0.11.0,
ruby-lsp-rails < 0.2.6 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.11.0.
And because ruby-lsp-rails >= 0.2.6, < 0.2.7 depends on ruby-lsp >= 0.11.0, <
0.12.0
  and ruby-lsp-rails >= 0.2.7, < 0.2.8 depends on ruby-lsp >= 0.12.0, < 0.13.0,
ruby-lsp-rails < 0.2.8 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.13.0.
And because ruby-lsp-rails >= 0.2.8, < 0.2.10 depends on ruby-lsp >= 0.13.0, <
0.14.0
  and ruby-lsp-rails >= 0.2.10, < 0.3.1 depends on ruby-lsp >= 0.14.0, < 0.15.0,
ruby-lsp-rails < 0.3.1 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.15.0.
And because ruby-lsp-rails >= 0.3.1, < 0.3.4 depends on ruby-lsp >= 0.14.2, <
0.15.0
  and ruby-lsp-rails >= 0.3.4, < 0.3.6 depends on ruby-lsp >= 0.16.0, < 0.17.0,
ruby-lsp-rails < 0.3.6 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.15.0 OR >= 0.16.0, < 0.17.0.
And because ruby-lsp-rails >= 0.3.6, < 0.3.7 depends on ruby-lsp >= 0.16.5, <
0.17.0
  and ruby-lsp-rails >= 0.3.7, < 0.3.8 depends on ruby-lsp >= 0.17.0, < 0.18.0,
ruby-lsp-rails < 0.3.8 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.15.0 OR >= 0.16.0, < 0.18.0.
And because ruby-lsp-rails >= 0.3.8, < 0.3.12 depends on ruby-lsp >= 0.17.2, <
0.18.0
and ruby-lsp-rails >= 0.3.12, < 0.3.16 depends on ruby-lsp >= 0.17.12, <
0.18.0,
ruby-lsp-rails < 0.3.16 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.15.0 OR >= 0.16.0, < 0.18.0.
And because ruby-lsp-rails >= 0.3.16, < 0.3.17 depends on ruby-lsp >= 0.18.0, <
0.19.0
and ruby-lsp-rails >= 0.3.17, < 0.3.19 depends on ruby-lsp >= 0.19.0, <
0.20.0,
ruby-lsp-rails < 0.3.19 requires ruby-lsp ~> 0.4.5 OR >= 0.6.2, < 0.9.0 OR >=
0.9.1, < 0.15.0 OR >= 0.16.0, < 0.20.0.
And because ruby-lsp-rails >= 0.3.19, < 0.3.22 depends on ruby-lsp >= 0.20.0, <
0.21.0
and ruby-lsp >= 0.4.5, < 0.5.A OR >= 0.6.2, < 0.9.0 OR >= 0.9.1, < 0.15.0 OR
>= 0.16.0, < 0.21.0 could not be found in
https://github.com/Shopify/ruby-lsp.git (at main@fc2d91c),
  ruby-lsp-rails < 0.3.22 cannot be used.
So, because ruby-lsp-rails >= 0.3.22 depends on ruby-lsp >= 0.21.2, < 0.22.0
and ruby-lsp >= 0.21.2, < 0.22.0 could not be found in
https://github.com/Shopify/ruby-lsp.git (at main@fc2d91c),
  version solving has failed.

2024-11-20 15:33:56.245 [info] (trigo-app) /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/source/git/git_proxy.rb:354:in `allowed_with_path': The git source https://github.com/Shopify/ruby-lsp.git is not yet checked out. Please run `bundle install` before trying to start your application (Bundler::GitError)
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/source/git/git_proxy.rb:78:in `revision'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/source/git.rb:242:in `revision'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/source/git.rb:116:in `install_path'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/source/path.rb:134:in `expanded_path'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/source/path.rb:175:in `load_spec_files'

2024-11-20 15:33:56.245 [info] (trigo-app) 	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/source/git.rb:219:in `load_spec_files'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/source/path.rb:108:in `local_specs'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/source/git.rb:185:in `specs'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/source.rb:60:in `spec_names'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/source_map.rb:21:in `block in all_requirements'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/source_map.rb:20:in `map'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/source_map.rb:20:in `all_requirements'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/definition.rb:957:in `source_requirements'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/definition.rb:569:in `resolution_packages'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/definition.rb:551:in `resolver'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/definition.rb:626:in `start_resolution'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/definition.rb:311:in `resolve'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/definition.rb:579:in `materialize'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/definition.rb:203:in `specs'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/definition.rb:270:in `specs_for'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/runtime.rb:18:in `setup'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler.rb:162:in `setup'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/setup.rb:26:in `block in <top (required)>'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/ui/shell.rb:159:in `with_level'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/ui/shell.rb:111:in `silence'
	from /Users/derek/.rvm/gems/ruby-3.3.5/gems/bundler-2.5.9/lib/bundler/setup.rb:26:in `<top (required)>'
	from <internal:/Users/derek/.rvm/rubies/ruby-3.3.5/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
	from <internal:/Users/derek/.rvm/rubies/ruby-3.3.5/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
	from /Users/derek/.rvm/rubies/ruby-3.3.5/lib/ruby/3.3.0/rubygems.rb:1395:in `<top (required)>'
	from <internal:gem_prelude>:2:in `require'
	from <internal:gem_prelude>:2:in `<internal:gem_prelude>'

2024-11-20 15:33:56.250 [info] (trigo-app) [Error - 3:33:56 PM] Server initialization failed.
2024-11-20 15:33:56.250 [info] (trigo-app)   Message: Pending response rejected since connection got disposed
  Code: -32097 
2024-11-20 15:33:56.250 [info] (trigo-app) [Error - 3:33:56 PM] Ruby LSP client: couldn't create connection to server.
2024-11-20 15:33:56.250 [info] (trigo-app)   Message: Pending response rejected since connection got disposed
  Code: -32097 
2024-11-20 15:33:56.250 [error] (trigo-app) Error starting the server: Pending response rejected since connection got disposed
2024-11-20 15:33:56.250 [info] (trigo-app) [Error - 3:33:56 PM] Server process exited with code 1.
```

**Comment by vinistock** [comment]
Thank you for the report!

Your Ruby LSP settings are pointing the gem to the `main` branch. We made breaking changes as part of the `v0.22` release and those are still being addressed in the Rails add-on.

That means that there's no possible version that will satisfy dependency constraints until we cut a release for `ruby-lsp-rails`, which is why it's breaking.

We're working on addressing the breaking changes to cut a release, but if you point the gem to `main` this will always happen whenever there's a breaking change released. FWIW, the `branch` setting is more tailored for when we're testing very disruptive changes in a branch before releasing it to everybody. If you remove the `rubyLsp.branch` setting and reload VS Code, the problem should go away.

But I'm also curious as to why you configured the branch to main. Was the intention to get access to new features before they were released?

The reason I'm asking is because I've wondered for a while if we should do nightly beta releases. We could then provide a setting for beta testers to get those early versions installed.

**Comment by dmrwebdev** [comment]
Thanks for your quick response! Great catch thank you, I totally forgot I had set that awhile ago 🤦‍♂️

As for why I did that a few months back I was experiencing some severe frequent crashing of the language server whenever it tried to auto-format on save. Pointing the branch to main resolved the issue and I promptly forgot about it and never switched back lol.

Confirmed removing that setting has immediately resolved my issue, thanks for your help!
